src/ranged.cpp

```diff
@@ -1222,28 +1222,27 @@ static projectile make_gun_projectile( const item &gun ) {
     proj.speed  = 1000;
     proj.impact = damage_instance::physical( 0, gun.gun_damage(), 0, gun.gun_pierce() );
     proj.range = gun.gun_range();
+    proj.proj_effects = gun.ammo_effects();
 
-    const auto curammo = gun.ammo_data();
-
-    // Consider both effects from the gun and ammo
     auto &fx = proj.proj_effects;
-    fx.insert( gun.type->gun->ammo_effects.begin(), gun.type->gun->ammo_effects.end() );
-    fx.insert( curammo->ammo->ammo_effects.begin(), curammo->ammo->ammo_effects.end() );
 
-    if( curammo->phase == LIQUID || fx.count( "SHOT" ) || fx.count("BOUNCE" ) ) {
+    if( ( gun.ammo_data() && gun.ammo_data()->phase == LIQUID ) ||
```

Comment:
It may be a good idea to have a default `ammo_data` now, to avoid checking for `nullptr`.